### PR TITLE
Fix SSS radius bug (+ misc changes)

### PR DESCRIPTION
**Pull request by EvgeniiG**
Can't seem to rebase my SSS changes on top of the older master, so have to include the volumetrics-related changes as well.

### CODE REVIEW

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/Common.hlsl [hunk 1]
```diff
@@ -108,6 +108,10 @@
 
 #endif // #ifndef real
 
+#ifdef  SHADER_STAGE_COMPUTE
+#define SHADER_TARGET 50 // Not defined otherwise
```
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/Common.hlsl]: shouldn't this be 45 ? Also need to check with graphic fondation team, looks like a "true" bug here

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDStringConstants.cs [hunk 2]
```diff
@@ -293,12 +293,12 @@ public static class HDShaderIDs
         public static readonly int _Result1 = Shader.PropertyToID("_Result1");
 
         public static readonly int _AmbientProbeCoeffs         = Shader.PropertyToID("_AmbientProbeCoeffs");
-        public static readonly int _GlobalFog_Extinction       = Shader.PropertyToID("_GlobalFog_Extinction");
-        public static readonly int _GlobalFog_Scattering       = Shader.PropertyToID("_GlobalFog_Scattering");
-        public static readonly int _GlobalFog_Asymmetry        = Shader.PropertyToID("_GlobalFog_Asymmetry");
+        public static readonly int _Global_Extinction          = Shader.PropertyToID("_Global_Extinction");
```
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDStringConstants.cs]: Follow convention : CameCase, no "_" in the middle.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Lighting/Deferred.shader [hunk 3]
```diff
@@ -124,7 +124,7 @@ Shader "Hidden/HDRenderPipeline/Deferred"
                 Outputs outputs;
 
             #ifdef OUTPUT_SPLIT_LIGHTING
-                if (_EnableSubsurfaceScattering != 0 && PixelHasSubsurfaceScattering(bsdfData))
+                if (HasFeatureFlag(bsdfData.materialFeatures, MATERIAL_FEATURE_FLAGS_SSS_OUTPUT_SPLIT_LIGHTING))
```
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Lighting/Deferred.shader]: No, we don't have the right to access bsdfData here. You must use a function, simialr  to PixelHasSubsurfaceScattering

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Lighting/LightLoop/Deferred.compute [hunk 4]
```diff
@@ -153,7 +153,7 @@ void SHADE_OPAQUE_ENTRY(uint2 dispatchThreadId : SV_DispatchThreadID, uint2 grou
     float3 specularLighting;
     LightLoop(V, posInput, preLightData, bsdfData, bakeLightingData, featureFlags, diffuseLighting, specularLighting);
 
-    if (_EnableSubsurfaceScattering != 0 && PixelHasSubsurfaceScattering(bsdfData))
+    if (HasFeatureFlag(bsdfData.materialFeatures, MATERIAL_FEATURE_FLAGS_SSS_OUTPUT_SPLIT_LIGHTING))
```
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Lighting/LightLoop/Deferred.compute]: See previous comment

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 5]
```diff
@@ -139,6 +139,10 @@ static const uint kFeatureVariantFlags[NUM_FEATURE_VARIANTS] =
     /* 26 */ LIGHT_FEATURE_MASK_FLAGS_OPAQUE | MATERIAL_FEATURE_MASK_FLAGS, // Catch all case with MATERIAL_FEATURE_MASK_FLAGS is needed in case we disable material classification
 };
 
+#define MATERIAL_FEATURE_FLAGS_SSS_OUTPUT_SPLIT_LIGHTING         ((MATERIAL_FEATURE_MASK_FLAGS + 1) << 0)
```
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: Add comment about that we going to merge inside materialfeatures other stuff and use bitfield as an optimization for GCN

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl [hunk 6]
```diff
@@ -229,7 +233,8 @@ void FillMaterialSSS(uint diffusionProfile, float subsurfaceMask, inout BSDFData
     bsdfData.diffusionProfile = diffusionProfile;
     bsdfData.fresnel0 = _TransmissionTintsAndFresnel0[diffusionProfile].a;
     bsdfData.subsurfaceMask = subsurfaceMask;
-    // Note: ApplySubsurfaceScatteringTexturingMode also test the diffusionProfile for updating diffuseColor based on SSS
+    bsdfData.materialFeatures |= (_EnableSubsurfaceScattering != 0) ? MATERIAL_FEATURE_FLAGS_SSS_OUTPUT_SPLIT_LIGHTING : 0;
```
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: add comment here

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/ShaderPass/ShaderPassForward.hlsl [hunk 7]
```diff
@@ -82,7 +82,7 @@ void Frag(PackedVaryingsToPS packedInput,
         LightLoop(V, posInput, preLightData, bsdfData, bakeLightingData, featureFlags, diffuseLighting, specularLighting);
 
 #ifdef OUTPUT_SPLIT_LIGHTING
-        if (_EnableSubsurfaceScattering != 0 && PixelHasSubsurfaceScattering(bsdfData))
+        if (HasFeatureFlag(bsdfData.materialFeatures, MATERIAL_FEATURE_FLAGS_SSS_OUTPUT_SPLIT_LIGHTING))
```
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/HDRenderPipeline/HDRP/ShaderPass/ShaderPassForward.hlsl]: See previous comment

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/Common.hlsl (hunk 1 above)
Comment (EvgeniiG) [reply ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/Common.hlsl]: Our shader compiler compiles compute shaders for 5.0.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDStringConstants.cs (hunk 2 above)
Comment (EvgeniiG) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/HDStringConstants.cs]: Good point, will fix this.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Lighting/Deferred.shader (hunk 3 above)
Comment (EvgeniiG) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Lighting/Deferred.shader]: Will fix this.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Lighting/LightLoop/Deferred.compute (hunk 4 above)
Comment (EvgeniiG) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Lighting/LightLoop/Deferred.compute]: Will fix this.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl (hunk 5 above)
Comment (EvgeniiG) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/Material/Lit/Lit.hlsl]: Will add a comment. The optimization is not GCN-specific.

File: ScriptableRenderPipeline/HDRenderPipeline/HDRP/ShaderPass/ShaderPassForward.hlsl (hunk 7 above)
Comment (EvgeniiG) [reply ScriptableRenderPipeline/HDRenderPipeline/HDRP/ShaderPass/ShaderPassForward.hlsl]: Will fix this.

File: ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/Common.hlsl [hunk 8]
```diff
@@ -108,6 +108,18 @@
 
 #endif // #ifndef real
 
+// Target in compute shader are supported in 2018.2, for now define ours
```
Comment (sebastienlagarde) [inline ScriptableRenderPipeline/Core/CoreRP/ShaderLibrary/Common.hlsl]: Note: See the change for SHADER_TARGET

**Pull request merged**